frontend/server/src/main/java/org/pytorch/serve/metrics/Metric.java

```diff
@@ -41,13 +50,15 @@ public Metric(
             String metricName,
             String value,
             String unit,
+            String type,
             String hostName,
             Dimension... dimensions) {
         this.metricName = metricName;
         this.value = value;
         this.unit = unit;
+        this.type = type;
         this.hostName = hostName;
-        this.dimensions = Arrays.asList(dimensions);
+        this.setDimensions(Arrays.asList(dimensions));
```

Comment:
Why original "this.dimensions = Arrays.asList(dimensions);" needs to be replaced with setDimensions?

Reply:
This is because [`setDimensions`](https://github.com/pytorch/serve/pull/2769/files#diff-fb226a2a07fa69a6476ec5fe0cb7467c3e5b82ba1193668127b88bc46e400963R126-R134) now takes care of setting up the values of `dimensionNames` and `dimensionsValues` as well to support the `getDimensionNames` and `getDimensionValues` methods.

Reply:
got it. Thanks